makelib/golang.mk

```diff
@@ -217,60 +193,49 @@ go.fmt.simplify: $(GOFMT)
 
 go.validate: go.vet go.fmt
 
-ifeq ($(GO111MODULE),on)
-
-go.vendor.lite go.vendor.check:
-	@$(INFO) verify dependencies have expected content
-	@$(GOHOST) mod verify || $(FAIL)
-	@$(OK) go modules dependencies verified
-
-go.vendor.update:
-	@$(INFO) update go modules
-	@$(GOHOST) get -u ./... || $(FAIL)
-	@$(OK) update go modules
+go.vendor.lite go.vendor.check: go.modules.check
+go.vendor.update: go.modules.update
+go.vendor: go.modules.download
 
-go.vendor:
-	@$(INFO) go mod vendor
-	@$(GOHOST) mod vendor || $(FAIL)
-	@$(OK) go mod vendor
+go.modules.check: go.modules.tidy.check go.modules.verify
 
-else
+go.modules.download:
+	@$(INFO) mod download
+	@$(GO) mod download || $(FAIL)
+	@$(OK) mod download
 
-go.vendor.lite: $(DEP)
-#	dep ensure blindly updates the whole vendor tree causing everything to be rebuilt. This workaround
-#	will only call dep ensure if the .lock file changes or if the vendor dir is non-existent.
-	@if [ ! -d $(GO_VENDOR_DIR) ]; then \
-		$(MAKE) vendor; \
-	elif ! $(DEP) ensure -no-vendor -dry-run &> /dev/null; then \
-		$(MAKE) vendor; \
-	fi
+go.modules.verify:
+	@$(INFO) verify go modules dependencies have expected content
+	@$(GO) mod verify || $(FAIL)
+	@$(OK) go modules dependencies verified
 
-go.vendor.check: $(DEP)
-	@$(INFO) checking if vendor deps changed
-	@$(DEP) check -skip-vendor || $(FAIL)
-	@$(OK) vendor deps have not changed
+go.modules.tidy:
+	@$(INFO) mod tidy
+	@$(GO) mod tidy
+	@$(OK) mod tidy
 
-go.vendor.update: $(DEP)
-	@$(INFO) updating vendor deps
-	@$(DEP) ensure -update -v || $(FAIL)
-	@$(OK) updating vendor deps
+go.modules.tidy.check:
+	@$(INFO) verify go modules dependencies are tidy
+	@$(GO) mod tidy
+	@$(shell git diff --exit-code --name-only go.mod go.sum)
+	@$(OK) go modules are tidy
 
-go.vendor: $(DEP)
-	@$(INFO) dep ensure
-	@$(DEP) ensure || $(FAIL)
-	@$(OK) dep ensure
+go.modules.update:
+	@$(INFO) update go modules
+	@$(GO) get -u ./... || $(FAIL)
+	@$(MAKE) go.modules.tidy
+	@$(MAKE) go.modules.verify
+	@$(OK) update go modules
 
-endif
+go.modules.clean:
+	@$(GO) clean -modcache
 
 go.clean:
-	@# `go modules` creates read-only folders under WORK_DIR
-	@# make all folders within WORK_DIR writable, so they can be deleted
-	@if [ -d $(WORK_DIR) ]; then chmod -R +w $(WORK_DIR); fi
-
+	@$(GO) clean -cache -testcache -modcache
 	@rm -fr $(GO_BIN_DIR) $(GO_TEST_DIR)
 
 go.distclean:
-	@rm -rf $(GO_VENDOR_DIR) $(GO_PKG_DIR)
```

Comment:
Do we still need this target if its simply echo'ing its status?

Reply:
I can remove it, did not want to break any existing repos/CI pipelines that might be relying on it

Reply:
Makes sense. I'd lean towards removing it since it's a no-op at this point.